# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1296511764 18000
#      Mon Jan 31 17:09:24 2011 -0500
# Branch release-3-4-x
# Node ID 1c4cad056b7981471ea3d82b137522c58b3affd5
# Parent  a85d14454adc9585963dbbe300943bf0f0b48bfb
sparse matrix indexed assignment bug

diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,14 @@
+2011-01-31  John W. Eaton  <jwe@octave.org>
+
+	* Sparse.cc (Sparse<T>::assign (const idx_vector&, const idx_vector&,
+	const Sparse<T>&)): Handle case of LHS originally empty, and
+	with non-colon indices.  Fix typo in previous change.  Bug #32263.
+
 2011-01-30  Pascal Dupuis  <Pascal.Dupuis@worldonline.be>
 
 	* lo-sysdep.cc (opendir): On error, free allocated DIR object
 	before returning.
 
 2011-01-28  John W. Eaton  <jwe@octave.org>
 
 	* eigs-base.cc: Remove #endif corresponding to "#ifdef HAVE_ARPACK",
diff --git a/liboctave/Sparse.cc b/liboctave/Sparse.cc
--- a/liboctave/Sparse.cc
+++ b/liboctave/Sparse.cc
@@ -1870,25 +1870,30 @@ Sparse<T>::assign (const idx_vector& idx
 
       if (orig_zero_by_zero)
         {
           if (idx_i.is_colon ())
             {
               nrx = n;
 
               if (idx_j.is_colon ())
-                ncx = n;
+                ncx = m;
               else
                 ncx = idx_j.extent (nc);
             }
           else if (idx_j.is_colon ())
             {
               nrx = idx_i.extent (nr);
               ncx = m;
             }
+          else
+            {
+              nrx = idx_i.extent (nr);
+              ncx = idx_j.extent (nc);
+            }
         }
       else
         {
           nrx = idx_i.extent (nr);
           ncx = idx_j.extent (nc);
         }
 
       // Try to resize first if necessary.
